### ISSUE TITLE
validate or auto-load IDs for builtins/nodes from modern.h

### ISSUE THREAD

**Issue by tinyplasticgreyknight** [tinyplasticgreyknight/modern-docs#5]

**Comment by IreneKnapp** [comment]
Oh, neat. :)  My observation here is that the number should be the primary key; since names aren't part of the format, it's conceivable they could change someday, and certainly will in these early draft stages.  I don't think of this as a high-priority feature but it would definitely be neat to have your build code check modern.h for the current name of each entity that has a doc entry, and update the docs to match that.

The workflow I'd imagine for this feature is something like:

``` bash
$ python make-rest.py rstsource
Error: Names don't match for modern_node 44; docs say lambda and header says lambda_value.
You may fix this manually, or invoke python make-rest.py --update-identifiers rstsource.

$ python make-rest.py --update-identifiers rstsource
Changed line ... of ... from lambda to lambda_value.

$ git diff
... etc
```

In other words, it validates, and refuses to proceed if it's invalid, and offers to fix things for you but doesn't proceed unless you tell it to.  That way the human stays responsible for version-control operations.

This might be a lot of work, and please don't feel obligated!  It's just how I'd design the feature in case you wanted that input.

**Comment by tinyplasticgreyknight** [comment]
Another option might be to just not list the names in the YAML files at all, only the numeric ID.  The name is then found automatically by inspecting the header file.  This makes the YAML a bit uglier though, so validation might be a better move.

**Comment by IreneKnapp** [comment]
Yeah, I could go either way on that.  Having the names there makes it easier to know what you're doing if you need to edit the structure, so I'd lean in favor of keeping them but validating them.
